pr/game.py

```diff
@@ -1,150 +1,184 @@
+        self.__players['o'] = input('Input second player name (it will play with o): ')
+
+    @staticmethod
+    def __get_value(message, down_board, up_board=20):
+        is_incorrect_input = True
+        player_input = 0
+        while is_incorrect_input:
+            try:
+                player_input = int(input(message))
+            except ValueError:
+                print('Incorrect input!')
+            else:
+                if player_input < down_board or player_input > up_board:
+                    print('Incorrect value!')
+                else:
+                    is_incorrect_input = False
 
+        return player_input
 
-def get_next_char():
-    global is_first_player, next_char
-    if is_first_player:
-        next_char = 'x'
-    else:
-        next_char = 'o'
-    is_first_player = not is_first_player
-    return next_char
+    def __get_next_char(self):
+        if self.__is_first_player:
+            next_char = 'x'
+        else:
+            next_char = 'o'
+        self.__is_first_player = not self.__is_first_player
+        self.__next_char = next_char
+        return next_char
+
+    def __create_board(self, size=3):
+        for i in range(size):
+            self.__board.append([' '] * size)
+        return self.__board
+
+    def __next_step(self):
+        print('Step of ' + self.__players[self.__next_char])
+        row = self.__get_value('Enter row position from 1 till ' + str(len(self.__board)) + ': ', 1,
+                               len(self.__board)) - 1
+        column = self.__get_value('Enter column position from 1 till ' + str(len(
+            self.__board)) + ': ', 1, len(self.__board)) - 1
+        return column, row
+
+    def __set_cell_value(self, value, column, row):
+        if self.__check_cell(column, row):
+            self.__set_cell(value, column, row)
+            self.__cell_counter -= 1
+        else:
+            self.__get_next_char()
+            print('Wrong position!')
 
+    def __check_cell(self, column, row):
+        return self.__board[column][row] == ' '
 
-def create_board(size=3):
-    for i in range(size):
-        board.append([' '] * size)
-    return board
+    def __set_cell(self, value, column, row):
+        self.__board[column][row] = value
 
+    def __is_round_over(self, char, column, row):
+        return self.__check_lines(char, column, row) \
+               or self.__check_diagonals(char, column, row) or self.__check_full()
 
-def next_step():
-    print('Next step of ' + players[next_char])
-    row = get_meaning('Enter row position from 1 till ' + str(len(board)) + ': ', 1, len(board)) - 1
-    column = get_meaning('Enter column position from 1 till ' + str(len(board)) + ': ', 1,
-                         len(board)) - 1
-    return column, row
+    def __check_lines(self, char, column, row):
+        result_column = self.__check_column(char, column)
+        result_row = self.__check_row(char, row)
 
+        if result_column or result_row:
```

Comment:
Why do we need this check and log in several places? 
It could be done only once in __is_round_over()

Reply:
Fixed